src/backend/gsroa.c

```diff
@@ -198,6 +270,13 @@ void sliceStructs()
             sia2[si + n].canSlice = false;
             if (sia[si].canSlice)
             {
+                if (!sia[si].accessSlice)
+                {
+                    // If never did access it as a slice, don't slice
+                    sia[si].canSlice = false;
```

Comment:
I prefer `can` to be answering a question, and `do` to actually perform the work.